compiler/modules/types/list.bal

```diff
@@ -126,17 +142,65 @@ function listSubtypeIsEmpty(Context cx, SubtypeData t) returns boolean {
     return isEmpty;    
 }
 
+function listMemberAt(SemType[] members, int i) returns SemType {
+    return members[int:min(i, members.length() - 1)];
+}
+
+function listDecompressMembersForSet(int setIndex, SemType[] members, int repeatCount, SemType withNewMember) returns int {
+    int memberLen = members.length();
+    boolean lastMemberRepeats = repeatCount != 0;
+    if lastMemberRepeats {        
```

Comment:
We could merge this `if-else` block and the `if-else` block below this. I felt that keeping the early return bits separate will make the method cleaner.